### PR TITLE
Try to create a type description by loading a class if the pool fails

### PR DESCRIPTION
**Pull request by bantonsson**

### CODE REVIEW

File: dd-java-agent/agent-tooling/src/main/java/datadog/trace/agent/tooling/bytebuddy/DDCachingPoolStrategy.java [hunk 1]
```diff
@@ -248,12 +256,52 @@ public void clear() {
...
+
+    @Override
+    public TypeDescription resolve() {
+      // Intentionally not "thread safe". Duplicate work deemed an acceptable trade-off.
+      if (!isResolved) {
+        Class<?> klass = null;
+        ClassLoader classLoader = loaderRef.get();
+        if (classLoader != null) {
+          try {
+            klass = classLoader.loadClass(className.replace('.', '\\'));
```
Comment (richardstartin) [inline dd-java-agent/agent-tooling/src/main/java/datadog/trace/agent/tooling/bytebuddy/DDCachingPoolStrategy.java:284]: Shouldn't this be `klass = classLoader.loadClass(className.replace('.', '/'));` ?

Comment (bantonsson) [reply dd-java-agent/agent-tooling/src/main/java/datadog/trace/agent/tooling/bytebuddy/DDCachingPoolStrategy.java:284]: Doh! Thanks, and the test is without a package name... 😔

Comment (bantonsson) [reply dd-java-agent/agent-tooling/src/main/java/datadog/trace/agent/tooling/bytebuddy/DDCachingPoolStrategy.java]: There was even a double error, and there should be no `replace`.

**Review by richardstartin — APPROVED** [review]

File: dd-java-agent/agent-tooling/src/main/java/datadog/trace/agent/tooling/bytebuddy/DDCachingPoolStrategy.java [hunk 2]
```diff
@@ -248,12 +256,52 @@ public void clear() {
...
+
+    @Override
+    public TypeDescription resolve() {
+      // Intentionally not "thread safe". Duplicate work deemed an acceptable trade-off.
+      if (!isResolved) {
+        Class<?> klass = null;
+        ClassLoader classLoader = loaderRef.get();
+        if (classLoader != null) {
+          try {
+            klass = classLoader.loadClass(className);
```
Comment (tylerbenson) [inline dd-java-agent/agent-tooling/src/main/java/datadog/trace/agent/tooling/bytebuddy/DDCachingPoolStrategy.java:284]: Should we log a debug message when this happens?

**Review by tylerbenson — APPROVED** [review]
Nice strategy. I think it would be worth adding a comment explaining that classes loaded in such manner will be directly loaded and not allowed to be instrumented. I think this is an acceptable tradeoff, but worth noting.

File: dd-java-agent/agent-tooling/src/main/java/datadog/trace/agent/tooling/bytebuddy/DDCachingPoolStrategy.java (hunk 2 above)
Comment (bantonsson) [reply dd-java-agent/agent-tooling/src/main/java/datadog/trace/agent/tooling/bytebuddy/DDCachingPoolStrategy.java:284]: Yes, we should definitely log when we are able to load the class. I'll add a comment as well.

**Review by tylerbenson — APPROVED** [review]

**Pull request merged**